File: 01. Leetcode/faangdsa/sachin/arrays/19/findPivotIndex.ts
function pivotIndex(nums: number[]): number {
    let l = 0;
    let r = 0;

    for (let num of nums){
        r += num;
    }

    for (let i=0; i<nums.length; i++){
        r-=nums[i];
        if (l == r){
            return i;
        }
        l+=nums[i];
    }
    return -1;
};

File: 01. Leetcode/faangdsa/sachin/arrays/19/find_pivot_index.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        l, r = 0, sum(nums)
        for idx, num in enumerate(nums):
            r-=num
            if l==r:
                return idx
            l+=num
        return -1
        